File: examples/c++/uln200xa.cxx
/*
 * Author: Jon Trulson <jtrulson@ics.com>
 * Copyright (c) 2015 Intel Corporation.
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */

#include <iostream>

#include "uln200xa.hpp"
#include "upm_utilities.h"

using namespace std;

int
main()
{
    //! [Interesting]

    // Instantiate a Stepper motor on a ULN200XA Darlington controller.

    // This was tested with the Grove Gear Stepper Motor with Driver

    // Wire the pins so that I1 is pin D8, I2 is pin D9, I3 is pin D10 and
    // I4 is pin D11
    upm::ULN200XA uln200xa(4096, 8, 9, 10, 11);

    uln200xa.setSpeed(5);
    uln200xa.setDirection(ULN200XA_DIR_CW);
    cout << "Rotating 1 revolution clockwise." << endl;
    uln200xa.stepperSteps(4096);
    cout << "Sleeping for 2 seconds..." << endl;
    upm_delay(2);
    cout << "Rotating 1/2 revolution counter clockwise." << endl;
    uln200xa.setDirection(ULN200XA_DIR_CCW);
    uln200xa.stepperSteps(2048);

    // turn off the power
    uln200xa.release();

    //! [Interesting]
    cout << "Exiting..." << endl;

    return 0;
}

File: examples/javascript/uln200xa.js
/*
* Author:  Jon Trulson <jtrulson@ics.com>
* Copyright (c) 2015-2016 Intel Corporation.
*
* This program and the accompanying materials are made available under the
* terms of the The MIT License which is available at
* https://opensource.org/licenses/MIT.
*
* SPDX-License-Identifier: MIT
*/

var Uln200xa_lib = require('jsupm_uln200xa');

// Instantiate a Stepper motor on a ULN200XA Darlington Motor Driver
// This was tested with the Grove Geared Step Motor with Driver

// Instantiate a ULN2003XA stepper object
var myUln200xa_obj = new Uln200xa_lib.ULN200XA(4096, 8, 9, 10, 11);

myUln200xa_obj.goForward = function()
{
    myUln200xa_obj.setSpeed(5); // 5 RPMs
    myUln200xa_obj.setDirection(Uln200xa_lib.ULN200XA_DIR_CW);
    console.log("Rotating 1 revolution clockwise.");
    myUln200xa_obj.stepperSteps(4096);
};

myUln200xa_obj.reverseDirection = function()
{
	console.log("Rotating 1/2 revolution counter clockwise.");
	myUln200xa_obj.setDirection(Uln200xa_lib.ULN200XA_DIR_CCW);
	myUln200xa_obj.stepperSteps(2048);
};

myUln200xa_obj.stop = function()
{
	myUln200xa_obj.release();
};

myUln200xa_obj.quit = function()
{
	myUln200xa_obj = null;
	Uln200xa_lib.cleanUp();
	Uln200xa_lib = null;
	console.log("Exiting");
	process.exit(0);
};

// Run ULN200xa driven stepper
myUln200xa_obj.goForward();
setTimeout(myUln200xa_obj.reverseDirection, 2000);
setTimeout(function()
{
	myUln200xa_obj.stop();
	myUln200xa_obj.quit();
}, 2000);
